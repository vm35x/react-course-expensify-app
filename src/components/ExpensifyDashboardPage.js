import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from './ExpensesSummary';

const ExpensifyDashboardPage = () => (
  <div>
  <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpensifyDashboardPage;
