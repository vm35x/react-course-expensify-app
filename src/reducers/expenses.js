//Expenses Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  //debugger;
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    //return state.concat(action.expense)
    case "REMOVE_EXPENSE":
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    case "SET_EXPENSES":
      return action.expenses;
    default:
      return state;
  }
};
