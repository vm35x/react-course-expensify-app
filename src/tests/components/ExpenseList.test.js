import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test("should render expense list on render", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render expense list with empty message", () => {
  const expenses = [];
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});
