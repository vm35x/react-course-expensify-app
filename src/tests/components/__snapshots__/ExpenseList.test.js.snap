// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list on render 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={1195}
      createdAt={0}
      description="Rent"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={2222}
      createdAt={-345600000}
      description="Gas"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={44}
      createdAt={345600000}
      description="Food"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No expenses
      </span>
    </div>
  </div>
</div>
`;
